src/audit/github_env.rs

```diff
@@ -0,0 +1,56 @@
+use super::{audit_meta, WorkflowAudit};
+use crate::finding::{Confidence, Finding, Severity};
+use crate::models::Step;
+use crate::state::AuditState;
+use github_actions_models::workflow::job::StepBody;
+use std::ops::Deref;
+
+pub(crate) struct GitHubEnv;
+
+audit_meta!(GitHubEnv, "github-env", "dangerous use of $GITHUB_ENV");
+
+impl GitHubEnv {
+    fn uses_github_environment(&self, run_step_body: &str) -> bool {
+        // In the future we can improve over this implementation,
+        // eventually detecting how $GITHUB_ENV is being used
+        // and returning an Option<Confidence> instead
+
+        run_step_body.contains("$GITHUB_ENV") || run_step_body.contains("${GITHUB_ENV}")
```

Comment:
Just thinking out loud: in terms of improving this, we could probably use a regexp like:

```
^.+\s*>>?\s*"?\$\{?GITHUB_ENV\}?"?.*$
```

That'll catch a bunch of common cases, but it's also a mess. So we can defer that for a follow-up.

Reply:
Done [here](https://github.com/woodruffw/zizmor/pull/192/commits/f222f8655f3f0d4a184982ef551822fbd9c07fea) with all other suggestions for docs!